test/infrastructure/inmemory/pkg/runtime/cache/sync.go

```diff
@@ -50,10 +50,8 @@ func (c *cache) startSyncer(ctx context.Context) error {
 		c.syncQueue.ShutDown()
 	}()
 
```

Comment:
@sivchari I took another look. Let's keep this but make it concurrency safe
l.53 
```go
var syncLoopStarted atomic.Bool
```
l.56
```go
syncLoopStarted.Store(true)
```
l.85
```go
if !syncLoopStarted.Load() {
```

Reply:
@SubhasmitaSw 
Thank you for commenting about it. Surely, it might be right, but I think it's not necessary. In l.83, we check if all workers starts by atomic.Load(&workers) < int64(c.syncConcurrency) and I  believe it's enough to achieve to block data race. Thanks.

Reply:
The check in l.83++ does not include the syncloop

Reply:
Thanks you for your comment. Sorry, I mentioned codes after changing.

syncLoopStarted is only used in codes below

```go
	go func() {
		log.Info("Starting sync loop")
		syncLoopStarted = true
		for {
			select {
			case <-time.After(c.syncPeriod / 4):
				c.syncGroup(ctx)
			case <-ctx.Done():
				return
			}
		}
	}()
```

```go
	if err := wait.PollUntilContextTimeout(ctx, 50*time.Millisecond, 5*time.Second, false, func(context.Context) (done bool, err error) {
		if !syncLoopStarted {
			return false, nil
		}
		return true, nil
	}); err != nil {
		return fmt.Errorf("failed to start sync loop: %v", err)
	}
```

c.syncGroup is originally concurrency program since using goroutine. So it depends on the timing of Go runtime to change syncLoopStarted from false to true. In other words, we don't originally consider the timing when this variable is true. Thus,  I think the syncLoopStarted is unnecessary. We can keep concurrency safe [here](https://github.com/kubernetes-sigs/cluster-api/pull/10899/files/4d91c48876a0b63096767e18aec983f1ce642160#diff-3b18df68234d2162fd5983124b41008f8ff6c76c0071b7a309210226cd85b67fL93-L101).
What do you think ?

Reply:
I'm no sure I understand what you are saying. Yes the old code was not correctly waiting for syncLoopStarted to be true. So it might not be that important. But I would prefer cleanly waiting for the "syncLoop" go routine to be started compared to not waiting for it.

Reply:
What I wanted to say is that syncLoopStarted might we can be worth removing. But when I confirmed your previous comment, I'd decide to keep this variable while changing the type to atomic.Bool. Thanks.